### PR TITLE
[docs] Assertions documentation

### PR DESCRIPTION
**Pull request by helen-dikareva**
/cc @DevExpress/testcafe-docs

### CODE REVIEW

File: docs/articles/documentation/test-api/assertions/index.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+To check if the state of the tested webpage matches the one you expect to see, use *assertions*.
+
+TestCafe provides a comprehensive [set of assertions](assertion-api.md) and all of them based on Behavior Driven Development style (BDD-style).
+
+This topic contains the following sections.
+
+* [Assertion timeout](#assertion-timeout)
+* [Assertion options](#assertion-options)
+  * [options.timeout](#optionstimeout)
+
```
Comment (AlexanderMoskovkin) [inline docs/articles/documentation/test-api/assertions/index.md]: It seems strange to me that we write 
`This topic contains the following sections.`
... and there are no basic section.

Maybe it's better to do like in [Selectors article](https://devexpress.github.io/testcafe/documentation/test-api/selecting-page-elements/selectors.html)?

```md
* [Creating Assertions](#creating-assertions)
* [Assertion timeout](#assertion-timeout)
* [Assertion options](#assertion-options)
  * [options.timeout](#optionstimeout)
```

File: docs/articles/documentation/test-api/assertions/index.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+Parameter              | Type     | Description
+---------------------- | -------- | -------------------------------------------------------------------------------
+`actual`                 | Any type | The comparison value
+`expected`                 | Any type | Expected value
+`message`&#160;*(optional)* | String   | Assertion message, will be indicated in the report in case of error.
+`options`&#160;*(optional)* | Object   | See [Options](#assertion-options).
+
+Assertions allow you to check values of the standard types (String, Number, Object and etc.). You can also use assertions
+to check states of the webpage's elements.
+o check state of any element on the page in assertion, you can use properties and methods of
```
Comment (AlexanderMoskovkin) [inline docs/articles/documentation/test-api/assertions/index.md]: a typo: `To check...`

File: docs/articles/documentation/test-api/assertions/index.md (hunk 1 above)
Comment (AlexanderMoskovkin) [inline docs/articles/documentation/test-api/assertions/index.md]: /cc @lexkazakov

File: docs/articles/documentation/test-api/assertions/index.md [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+```
+
+Parameter              | Type     | Description
+---------------------- | -------- | -------------------------------------------------------------------------------
+`actual`                 | Any type | The comparison value
+`expected`                 | Any type | Expected value
+`message`&#160;*(optional)* | String   | Assertion message, will be indicated in the report in case of error.
+`options`&#160;*(optional)* | Object   | See [Options](#assertion-options).
+
+Assertions allow you to check values of the standard types (String, Number, Object and etc.). You can also use assertions
```
Comment (AlexanderMoskovkin) [inline docs/articles/documentation/test-api/assertions/index.md]: I think we should place `use assertions to check states ...` in the first place.
Maybe `Assertions allow you to check states of the webpage's elements or value of the standard types`
/cc @helen-dikareva @lexkazakov

File: docs/articles/documentation/test-api/assertions/index.md [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+to check states of the webpage's elements.
+o check state of any element on the page in assertion, you can use properties and methods of
+[DOM Node Snapshot](../selecting-page-elements/selectors.md#dom-node-snapshots)
+or use [Snapshot API Shorthands](../selecting-page-elements/selectors.md#snapshot-api-shorthands).
+
+The sample below demonstrates how to use assertions to check element states.
```
Comment (AlexanderMoskovkin) [inline docs/articles/documentation/test-api/assertions/index.md]: maybe `to check element state` ?

File: docs/articles/documentation/test-api/assertions/index.md [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+import { Selector } from 'testcafe';
+
+fixture `Example page`
+    .page `http://devexpress.github.io/testcafe/example/`;
+
+test('Check property of element', async t => {
+    const developerNameInput = Selector('#developer-name');
+
+    await t
+        .expect(developerNameInput().value).eql('', 'input is empty') // Assertion with DOM node snapshot property
```
Comment (AlexanderMoskovkin) [inline docs/articles/documentation/test-api/assertions/index.md]: Why not `.expect(developerNameInput.value)` ?
If you'd like to show how to check shapshot's property it should be:
```js
const input = await Selector('...')

t.expect(input.value)
```
But I prefer to show shorthand-way here

File: docs/articles/documentation/using-testcafe/command-line-interface.md [hunk 6]
```diff
@@ -280,6 +281,15 @@ attempts to obtain a node to be returned. See [Selector Timeout](../test-api/sel
...
 
+### --assertion-timeout \<ms\>
+
+Specifies the amount of time, in milliseconds, within which TestCafe makes attempts to successfully execute an [assertion](../test-api/assertions/index.md).
+See [Assertion Timeout](../test-api/assertions/index.md#assertion-timeout).
+
+```sh
+testcafe ie my-tests --assertion-timeout 10000
+```
+
```
Comment (AlexanderMoskovkin) [inline docs/articles/documentation/using-testcafe/command-line-interface.md]: I think we should add

**Default value**: `3000` like for the `speed` option

File: examples/use-assertions/test.js [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+
+test('Using DOM node snapshot', async t => {
+    const developerNameInput = Selector('#developer-name');
+    const initialValue       = await developerNameInput().value;
+
+    await t
+        .expect(initialValue).eql('', 'input is empty')
+        .typeText(developerNameInput, 'Peter Parker');
+
+    await t.expect(developerNameInput().value).contains('Parker', 'input has text');
```
Comment (AlexanderMoskovkin) [inline examples/use-assertions/test.js]: I think we should show directly how to use a snapshot in assertions. The common case: use get a snapshot a check several properties, like:
```js
var inputSnapshot = await Selector(...);

t
.expect(inputSnapshot.prop1).eql(value1)
.expect(inputSnapshot.prop2).notEq(value2);
```

File: docs/articles/documentation/test-api/assertions/index.md [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+
+This topic contains the following sections.
+
+* [Assertion timeout](#assertion-timeout)
+* [Assertion options](#assertion-options)
+  * [options.timeout](#optionstimeout)
+
+The following example demonstrates how to check whether two values are equal or not.
+
+```text
```
Comment (helen-dikareva) [inline docs/articles/documentation/test-api/assertions/index.md]: I think it should not.
Look at other examples:
text https://github.com/DevExpress/testcafe/pull/997/files#diff-70e36160da834fc09f9ab211e6fd2d9dR29
code  https://github.com/DevExpress/testcafe/pull/997/files#diff-70e36160da834fc09f9ab211e6fd2d9dR42

File: docs/articles/documentation/test-api/assertions/index.md (hunk 5 above)
Comment (helen-dikareva) [inline docs/articles/documentation/test-api/assertions/index.md]: 'Assertion with snapshot API shorthands' is below. Do you offer completely get rid of example with DOM node snapshot property?

File: docs/articles/documentation/using-testcafe/command-line-interface.md (hunk 6 above)
Comment (helen-dikareva) [inline docs/articles/documentation/using-testcafe/command-line-interface.md]: We didn't do it for `selector-timeout`. Should I add to both?

File: examples/use-assertions/test.js (hunk 7 above)
Comment (helen-dikareva) [inline examples/use-assertions/test.js]: Ok, you right!

File: docs/articles/documentation/test-api/assertions/index.md (hunk 2 above)
Comment (helen-dikareva) [inline docs/articles/documentation/test-api/assertions/index.md]: @lexkazakov

File: docs/articles/documentation/test-api/assertions/index.md (hunk 4 above)
Comment (helen-dikareva) [inline docs/articles/documentation/test-api/assertions/index.md]: @lexkazakov

File: docs/articles/documentation/test-api/assertions/index.md [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+    const developerNameInput = Selector('#developer-name');
+
+    await t
+        .expect(developerNameInput().value).eql('', 'input is empty') // Assertion with DOM node snapshot property
+        .typeText(developerNameInput, 'Peter')
+        .expect(developerNameInput.value).eql('Peter', 'input has text'); // Assertion with snapshot API shorthands
+});
+```
+
+## Assertion Timeout
```
Comment (kirovboris) [inline docs/articles/documentation/test-api/assertions/index.md]: We should specify the default timeout in this topic.

File: docs/articles/documentation/using-testcafe/command-line-interface.md [hunk 10]
```diff
@@ -280,6 +281,15 @@ attempts to obtain a node to be returned. See [Selector Timeout](../test-api/sel
 testcafe ie my-tests --selector-timeout 500000
 ```
 
+### --assertion-timeout \<ms\>
```
Comment (kirovboris) [inline docs/articles/documentation/using-testcafe/command-line-interface.md]: We should specify the default timeout in this topic.

File: docs/articles/index.html [hunk 11]
```diff
@@ -95,6 +95,34 @@
...
         <div class="feature">
             <span class="name">
+                Smart Assertions
+            </span>
+            <div class="description">
+                <p>TestCafe provides a set of <a href="documentation/test-api/assertions/assertion-api.html">built-in assertions</a>. Built-in assertions provide the following enhancements:</p>
+                <ul>
+                    <li>A full-featured set of built-in assertions, so you have no need to reference additional libraries.</li>
+                    <li>Easy to use assertions in test scenarios by chaining with test actions.</li>
+                    <li>Support for the <a href="documentation/test-api/selecting-page-elements/selectors.html#snapshot-api-shorthands">Snapshot API Shorthands</a>, that allows you to easily check states of page elements and use the Page Object methodology.</li>
```
Comment (kirovboris) [inline docs/articles/index.html]: state of page elements?

File: examples/use-assertions/test.js [hunk 12]
```diff
@@ -0,0 +1,35 @@
...
+
+const triedTestCafeCheckbox = Selector('#tried-test-cafe');
+
+test('Using snapshot API shorthands', async t => {
+    await t
+        .expect(triedTestCafeCheckbox.checked).notOk()
+        .click(triedTestCafeCheckbox)
+        .expect(triedTestCafeCheckbox.checked).ok();
+
+    var slider        = Selector('.ui-slider-handle');
```
Comment (kirovboris) [inline examples/use-assertions/test.js]: Let's use `let` and `const` in examples.

File: docs/articles/documentation/test-api/assertions/assertion-api.md [hunk 13]
```diff
@@ -0,0 +1,300 @@
...
+* [gte](#gte)
+* [lt](#lt)
+* [lte](#lte)
+* [within](#within)
+* [notWithin](#notwithin)
+
+## eql
+
+```text
+await t.expect(actual).eql(expected, message, options);
```
Comment (kirovboris) [inline docs/articles/documentation/test-api/assertions/assertion-api.md]: In other topics we use the style `await t.expect( actual ).eql( expected, message, options );` E.g. https://github.com/DevExpress/testcafe/blob/master/docs/articles/documentation/test-api/actions/drag-element.md

File: docs/articles/documentation/test-api/assertions/index.md (hunk 8 above)
Comment (AlexanderMoskovkin) [inline docs/articles/documentation/test-api/assertions/index.md]: Hmm., I see the same in Selectors. Ok.

File: docs/articles/documentation/test-api/assertions/index.md (hunk 5 above)
Comment (AlexanderMoskovkin) [inline docs/articles/documentation/test-api/assertions/index.md]: I think we should show the better way here. It's shorthand using. So, yes.

Meanwhile I don't know how does it work:
`t.expect(developerNameInput().value)`
If you want to show snapshot using, it should be `t.expect((await developerNameInput()).value)`. No?

File: docs/articles/documentation/using-testcafe/command-line-interface.md (hunk 6 above)
Comment (AlexanderMoskovkin) [inline docs/articles/documentation/using-testcafe/command-line-interface.md]: Yep

File: docs/articles/documentation/test-api/assertions/assertion-api.md [hunk 14]
```diff
@@ -0,0 +1,300 @@
...
+```
+
+Parameter              | Type                                              | Description
+---------------------- | ------------------------------------------------- | ------------------------------------------------------------------------------------------------------------------
+`actual`             | Any type | A comparison value
+`typeName`             | String | The expected type of an `actual` value
+`message`&#160;*(optional)* | String   | An assertion message that will be displayed in the report if the test fails
+`options`&#160;*(optional)* | Object   | See [Options](index.md#assertion-options).
+
+Asserts that type of `actual` is `typeName`. Type of `actual` will be determined as a constructor's name.
```
Comment (kirovboris) [inline docs/articles/documentation/test-api/assertions/assertion-api.md]: I'm still not sure about the phrase ```will be determined as a constructor's name.``` Can i use `number`, `Number`, `NUMBER`? I can't understand it from docs.

File: docs/articles/documentation/test-api/assertions/index.md [hunk 15]
```diff
@@ -0,0 +1,82 @@
...
+Parameter              | Type     | Description
+---------------------- | -------- | -------------------------------------------------------------------------------
+`actual`                 | Any type | The comparison value
+`expected`                 | Any type | Expected value
+`message`&#160;*(optional)* | String   | Assertion message, will be indicated in the report in case of error.
+`options`&#160;*(optional)* | Object   | See [Options](#assertion-options).
+
+Assertions allow you to check values of the standard types (String, Number, Object and etc.). You can also use assertions
+to check states of the webpage's elements.
+to check state of any element on the page in assertion, you can use properties and methods of
```
Comment (AlexanderMoskovkin) [inline docs/articles/documentation/test-api/assertions/index.md]: `To check... `?

File: docs/articles/documentation/test-api/assertions/index.md (hunk 1 above)
Comment (AlexanderMoskovkin) [inline docs/articles/documentation/test-api/assertions/index.md]: (approved by @lexkazakov) 
@helen-dikareva let's implement

File: docs/articles/documentation/test-api/assertions/index.md (hunk 3 above)
Comment (AlexanderMoskovkin) [inline docs/articles/documentation/test-api/assertions/index.md]: Use assertions to check states of webpage elements and any values of standard types (Strings,....)

File: docs/articles/documentation/test-api/assertions/assertion-api.md [hunk 16]
```diff
@@ -0,0 +1,300 @@
...
+
+```text
+await t.expect(actual).ok(message, options);
+```
+
+Parameter              | Type                                              | Description
+---------------------- | ------------------------------------------------- | ------------------------------------------------------------------------------------------------------------------
+`actual`             | Any type | A value that should be truthy
+`message`&#160;*(optional)* | String   | An assertion message that will be displayed in the report if the test fails
+`options`&#160;*(optional)* | Object   | See [Options](index.md#assertion-options).
```
Comment (churkin) [inline docs/articles/documentation/test-api/assertions/assertion-api.md]: Extra dot after "See Options".
I think.

File: docs/articles/index.html (hunk 11 above)
Comment (helen-dikareva) [inline docs/articles/index.html]: Tech writer says `states of page elements`

**Comment by helen-dikareva** [comment]
FPR

File: docs/articles/documentation/test-api/assertions/assertion-api.md (hunk 16 above)
Comment (lexkazakov) [inline docs/articles/documentation/test-api/assertions/assertion-api.md]: Everywhere in tables there must be dots in descriptions.

File: docs/articles/documentation/test-api/assertions/index.md [hunk 17]
```diff
@@ -0,0 +1,85 @@
...
+This topic contains the following sections.
+
+* [Using Assertions](#using-assertions)
+* [Assertion timeout](#assertion-timeout)
+* [Assertion options](#assertion-options)
+  * [options.timeout](#optionstimeout)
+
+## Using Assertions
+
+The following example demonstrates how to check whether two values are equal or not.
```
Comment (inikulin) [inline docs/articles/documentation/test-api/assertions/index.md]: I guess it shouldn't be an `example` here we demonstrate the general structure of assertions. Instead of `eql` we should use `[assertionType]` placeholder and link to the possible assertions types.

File: docs/articles/documentation/test-api/assertions/index.md [hunk 18]
```diff
@@ -0,0 +1,85 @@
...
+
+This topic contains the following sections.
+
+* [Using Assertions](#using-assertions)
+* [Assertion timeout](#assertion-timeout)
+* [Assertion options](#assertion-options)
+  * [options.timeout](#optionstimeout)
+
+## Using Assertions
+
```
Comment (inikulin) [inline docs/articles/documentation/test-api/assertions/index.md]: We should declare how our assertions are differ from the existing libraries from the start. We need to describe the motivation and mechanisms of our assertions here.

File: docs/articles/documentation/test-api/assertions/index.md [hunk 19]
```diff
@@ -0,0 +1,85 @@
...
+    const developerNameInput = Selector('#developer-name');
+
+    await t
+        .expect(developerNameInput.value).eql('', 'input is empty')
+        .typeText(developerNameInput, 'Peter')
+        .expect(developerNameInput.value).eql('Peter', 'input has text');
+});
+```
+
+## Assertion Timeout
```
Comment (inikulin) [inline docs/articles/documentation/test-api/assertions/index.md]: It's makes `options.timeout` ambiguous. Also it shouldn't be presented as `Assertion Timeout`. Our assertions mechanism should be described in details above.

File: docs/articles/index.html [hunk 20]
```diff
@@ -95,6 +95,33 @@
...
         <div class="feature">
             <span class="name">
+                Smart Assertions
+            </span>
+            <div class="description">
+                <p>TestCafe provides a set of <a href="documentation/test-api/assertions/assertion-api.html">built-in assertions</a>. Built-in assertions provide the following enhancements:</p>
+                <ul>
+                    <li>A full-featured set of built-in assertions means that do not need to reference additional libraries.</li>
+                    <li><a href="documentation/test-api/selecting-page-elements/selectors.html#snapshot-api-shorthands">Snapshot API Shorthands</a> allows you to easily check the states of page elements and use the Page Object methodology.</li>
+                    <li>By combining the Snapshot API Shorthands and <a href="documentation/test-api/assertions/index.html#assertion-options">configurable assertions</a>, you can avoid accidental errors and unstable tests</li>
```
Comment (inikulin) [inline docs/articles/index.html]: We should elaborate details about why it's cool and how it works.

File: examples/using-assertions/test.js [hunk 21]
```diff
@@ -0,0 +1,35 @@
...
+    const sliderTick    = Selector('.slider-value');
+    const initialOffset = await slider.offsetLeft;
+
+    await t
+        .dragToElement(slider, sliderTick.withText('9'))
+        .expect(slider.offsetLeft).gte(initialOffset);
+});
+
+test('Using DOM node snapshot', async t => {
+    const commentsTextArea = Selector('#comments');
```
Comment (inikulin) [inline examples/using-assertions/test.js]: We've decided to encourage people to not use snapshots there possible. So I guess it'll be better redesigned to use parametrized selector.

**Comment by inikulin** [comment]
@helen-dikareva There is a broken link on page

File: docs/articles/documentation/test-api/assertions/assertion-api.md [hunk 22]
```diff
@@ -0,0 +1,337 @@
...
+layout: docs
+title: Assertion methods
+permalink: /documentation/test-api/assertions/assertion-api.html
+checked: false
+---
+# Assertion API
+
+The following assertion methods are available.
+
+* [Equal](#equal)
```
Comment (inikulin) [inline docs/articles/documentation/test-api/assertions/assertion-api.md]: `Deep Equal`

File: docs/articles/documentation/test-api/assertions/assertion-api.md [hunk 23]
```diff
@@ -0,0 +1,337 @@
...
+* [Not Equal](#not-equal)
```
Comment (inikulin) [inline docs/articles/documentation/test-api/assertions/assertion-api.md]: Not Deep Equals

File: docs/articles/documentation/test-api/assertions/index.md [hunk 24]
```diff
@@ -0,0 +1,108 @@
...
+To make such tests stable, you need to add a timeout that will guarantee you a successful completion of required changes.
+Adding such timeouts can increase the test running time because of extra waiting.
+
+The main feature of TestCafe assertions is the use of built-in assertion query mechanism.
+To watch the state of web page elements, TestCafe provides [Selectors](../selecting-page-elements/selectors.md).
+If the TestCafe assertion receives a [Selector's property](../selecting-page-elements/selectors.md#snapshot-api-shorthands)
+as an actual value, TestCafe uses the smart assertion query mechanism:
+if an assertion did not passed, the test does not fail immediately. The assertion retries to pass multiple times and
+each time it re-requests the actual shorthand value. The test fails if the assertion could not complete successfully
+within a timeout.
```
Comment (inikulin) [inline docs/articles/documentation/test-api/assertions/index.md]: We need a code example here, e.g.:

*Page markup:*
```html
<div id="btn"></div>
<script>
var btn = document.getElementById('btn');

btn.addEventListener(function() { 
    window.setTimeout(function() {
        btn.innerText = 'Loading...';
    }, 100);
});
</script> 
```

*Test code:*
```js
test('Button click', async t => {
    const btn = Selector('#btn');

    await t
        .click(btn)
         // NOTE: Regular assertion will fail immediately, but TestCafe retries
         // DOM state assertions for some period of time, so this assertion will pass.
        .expect(btn.textContent).contains('Loading...'); 
});
```

File: docs/articles/documentation/test-api/assertions/index.md [hunk 25]
```diff
@@ -0,0 +1,108 @@
...
+To perform asynchronous functional tests, the additional timeout as usually added.
+
+![Asynchronous Functional Testing with Extra Waiting](../../../images/report.png)
+
+To make such tests stable, you need to add a timeout that will guarantee you a successful completion of required changes.
+Adding such timeouts can increase the test running time because of extra waiting.
+
+The main feature of TestCafe assertions is the use of built-in assertion query mechanism.
+To watch the state of web page elements, TestCafe provides [Selectors](../selecting-page-elements/selectors.md).
+If the TestCafe assertion receives a [Selector's property](../selecting-page-elements/selectors.md#snapshot-api-shorthands)
```
Comment (inikulin) [inline docs/articles/documentation/test-api/assertions/index.md]: We can completely remove two first sentences here: `The main feature of TestCafe assertions...` and `To watch the state of web page elements...`

Comment (inikulin) [inline docs/articles/documentation/test-api/assertions/index.md]: `Selector's DOM node state property`

File: docs/articles/index.html [hunk 26]
```diff
@@ -95,6 +95,28 @@
         </div>
         <div class="feature">
             <span class="name">
+                Smart Assertions
+            </span>
+            <div class="description">
+                <p>TestCafe provides a full-featured set of <a href="documentation/test-api/assertions/assertion-api.html">built-in assertions</a>, so you do not need to reference additional libraries. Everything you need at  your fingertips. The built-in assertions work fine with the <a href="documentation/test-api/selecting-page-elements/selectors.html#snapshot-api-shorthands">Selector Shorthands</a> and <a href="documentation/test-api/selecting-page-elements/selectors.html#dom-node-snapshots">Snapshots</a>. All of this allows you to use the <a href="documentation/test-api/assertions/index.html#smart-assertion-query-mechanism">Assertion Query Mechanism</a> and create stable, fast and reliable tests, that do not depend on page response speed</p>
```
Comment (inikulin) [inline docs/articles/index.html]: `Everything you need at  your fingertips.` - let's remove this sentence

Comment (inikulin) [inline docs/articles/index.html]: Instead of `The built-in assertions work fine with...` maybe rephrase it to `When used with Selector`s DOM node state properties assertions enable Smart Assertion Query Mechanism..`.

File: docs/articles/documentation/test-api/assertions/index.md [hunk 27]
```diff
@@ -0,0 +1,108 @@
...
+as an actual value, TestCafe uses the smart assertion query mechanism:
+if an assertion did not passed, the test does not fail immediately. The assertion retries to pass multiple times and
+each time it re-requests the actual shorthand value. The test fails if the assertion could not complete successfully
+within a timeout.
+
+![TestCafe Smart Assertion Query Mechanism](../../../images/report.png)
+
+The approach described above allows you to create stable tests that lack random errors.
+Using this approach, you can decrease the amount of time required to run all your tests without extra waitings.
```
Comment (inikulin) [inline docs/articles/documentation/test-api/assertions/index.md]: `The approach described above`, `Using this approach`. Rephrase into single sentence without awkward `this/that approach` references

Comment (lexkazakov) [inline docs/articles/documentation/test-api/assertions/index.md]: The approach described above allows you to create stable tests that lack random errors and run fast without extra waitings.

File: docs/articles/index.html (hunk 26 above)
Comment (lexkazakov) [inline docs/articles/index.html]: TestCafe provides a full-featured set of <a href="documentation/test-api/assertions/assertion-api.html">built-in assertions</a>, so you do not need to reference additional libraries. 
Use assertions with Selector's DOM node state properties to enable <a href="documentation/test-api/assertions/index.html#smart-assertion-query-mechanism">Smart Assertion Query Mechanism</a> and create stable, fast and reliable tests, that do not depend on page response time.</p>

File: docs/articles/documentation/test-api/assertions/index.md (hunk 24 above)
Comment (lexkazakov) [inline docs/articles/documentation/test-api/assertions/index.md]: **Example.**
Let's create a test for the following web page.
```html
<div id="btn"></div>
<script>
var btn = document.getElementById('btn');

btn.addEventListener(function() { 
    window.setTimeout(function() {
        btn.innerText = 'Loading...';
    }, 100);
});
</script> 
```
The test code will look as follows.

```js
test('Button click', async t => {
    const btn = Selector('#btn');

    await t
        .click(btn)
         // NOTE: Regular assertion will fail immediately, but TestCafe retries to run DOM state
         // assertions many times until this assertion pass successfully within the specified timeout.
        .expect(btn.textContent).contains('Loading...', {timeout:1000}); 
});
```

[Kazakov: pay attention to the specified timeout, that makes the example more clear]

**Comment by helen-dikareva** [comment]
FPR

File: docs/articles/documentation/getting-started/index.md [hunk 28]
```diff
@@ -149,32 +149,20 @@ A functional test also should check the result of actions performed.
 For example, the article header on the "Thank you" page should address a user by the entered name.
 To check if the header is correct, you have to add an assertion to the test.
 
-You can use assertions from Node's built-in [assert](https://nodejs.org/api/assert.html) module or from any third-party assertion library.
-Before calling assertions, make sure an assertion library is installed into your project.
-
-The following test demonstrates how to use an assertion from [Chai Assertion Library](http://chaijs.com/api/bdd/).
-Before running the test, install the assertion library by calling the `npm install --save-dev chai` command.
+he following test demonstrates how to use [build-in assertions](../test-api/assertions/index.md).
```
Comment (kirovboris) [inline docs/articles/documentation/getting-started/index.md]: typo

Comment (helen-dikareva) [inline docs/articles/documentation/getting-started/index.md]: thanks

**Comment by helen-dikareva** [comment]
FPR
